core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1041,10 +1041,22 @@ wasi_sock_addr_remote(wasm_exec_env_t exec_env, wasi_fd_t fd, uint8 *buf,
 }
 
 static wasi_errno_t
-wasi_sock_addr_resolve(wasm_exec_env_t exec_env, wasi_fd_t fd, const char *host,
-                       wasi_ip_port_t port, uint8 *buf, wasi_size_t size)
+wasi_addr_resolve(wasm_exec_env_t exec_env, const char *host,
```

Comment:
Not sure whether change the function name, isn't it related with socket? If yes, had better not change the name, and rename wasi_ssp_addr_resolve to wasi_ssp_sock_addr_resolve, and os_addr_resolve to os_sock_addr_resolve.

Reply:
The main reason I decided to rename this function was that the WASI function name that we export is `addr_resolve` (with no `socket_` prefix. Also, unlike other `socket_*` functions, this function does not operate on the socket that's being passed as argument. I see your point that it's part of the `socket` API so I'm not insisting on renaming it back to `socket_addr_resolve`, but in that case I think we should also change the name of the function in WASI to be consistent.

What do you think?

Reply:
Yes, had better change the related structure/function names to be `socket_` prefixed, it looks more clear, or it makes a little confused what wamr_addr_info_t means, bh_sock_addr_info_t or bh_sock_addr_t seems better.